### PR TITLE
Update UPE Intention with selected payment type

### LINKED ISSUE #2399: Pass the selected UPE payment method to the server when creating or updating the intent
Based on our discussion around transaction fees, it was discovered that we will need to pass along the selected payment method from the UPE to the server to allow the server to calculate the appropriate fees. The selected payment method can be determined [here](https://github.com/Automattic/woocommerce-payments/blob/eabe35cd95cebf04af1f431f14c18169a941d6b7/client/checkout/classic/upe.js#L215). And we will want to pass this along to the server as a string from the list (currently) of: `card, giropay, sofort`.

### PR DESCRIPTION
**Pull request by bborman22**
Fixes #2399 

#### Changes proposed in this Pull Request
This PR updates the UPE payment intention with the payment type the customer selected. Before this PR, the intention would contain an array of available payment methods based on the initial creation of the intention. This limited our ability to determine which method was ultimately selected and perform any additional processing on that selection. With this update, we will track which payment method was selected within the UPE by updating the `payment_method_types` to only include the selected payment method. One particular use case for this is around saving a payment method. Since some methods can't be saved, we know have some ability to perform this check server side before attempting to save the method and avoiding an error when updating the intention.

The approach was to add a hidden input to the checkout that tracks which method was selected. This field's value is then submitted along with the checkout process which updates the intention. There is also the direct AJAX call to update an intention that will now also handle this new field. 

If for some reason the selected payment method isn't successfully passed along, the `payment_method_types` remains unchanged and falls back to its current behavior.

#### Testing instructions

- Start with a store with this branch and UPE enabled.
- Use Euros on the store to allow CC, Giropay, and Sofort payment methods.
- Go to check out and select Giropay.
- Submit checkout form and approve.
- As admin, check the payment intention ID on the order and copy it's value.
- Within the Stripe dashboard, view the payment intention details, confirm that `payment_method_types` only contains Giropay.
- Perform the same steps except reject the Giropay request.
- Select CC after the rejection and perform a full checkout.
- Confirm in Stripe that the payment intention details only contains card.

- Also test this flow with the Order Pay page by manually creating and order and visiting the pay for order page.
- Checking throughout the code with breakpoints can also help determine that the `payment_method_types` is being updated properly.

-------------------

- [ ] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

**Review by FangedParakeet — APPROVED** [review]
I tested all the scenarios listed here--approving and failing payment using the regular checkout and the pay for order page--and everything worked out as expected for me! Code looks great to me too.

I noticed that this solves the problem of saving payments, while having payment methods that do not support saving payments enabled and I was consequently able to checkout and opt to save my card payment method at checkout even though Giropay was also present. However, even though the payment method saved correctly in my Stripe dashboard and the payment information appears to be present with correct details in my `wordpress` database, for some reason my newly minted payment method is not showing up on my account page. Anyways, I don't think that has anything to do with what's going on here, but let me know if any of these symptoms sound familiar or if anyone else has the cure for my malady.

In conclusion, this worked great and I approve.
